File: Backend/MetadataService/MetadataService.Core/DTOs/AddAudioDto.cs
using MetadataService.Core.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MetadataService.Core.DTOs
{
    public class AddAudioDto
    {
        public string? Title { get; set; }

        public string Description { get; set; } = "";

        public byte[]? Thumbnail { get; set; }

        public int DurationSec { get; set; }

        public long? FileSize { get; set; }

        public string? FileType { get; set; }

        public byte[]? FileChecksum { get; set; }

        public int? VisibilityId { get; set; }
    }
}

File: Frontend/audiowave6/lib/domain/dto/add_audio_dto.dart
import 'dart:typed_data';

class AddAudioDto {
  String? title;
  String description = "";
  Uint8List? thumbnail;
  int durationSec;
  int? fileSize;
  String? fileType;
  Uint8List? fileChecksum;
  int? visibilityId;
  List<String>? tags;

  AddAudioDto({
    this.title,
    this.description = "",
    this.thumbnail,
    required this.durationSec,
    this.fileSize,
    this.fileType,
    this.fileChecksum,
    this.visibilityId,
    this.tags,
  });
}
